lib/engine/activity_monitor.ts

```diff
@@ -24,47 +24,71 @@ import AssistantDispatcher from '../dialogue-agent/assistant_dispatcher';
 const DEFAULT_IDLE_TIMEOUT = 600000; // 10 minutes
 const DEFAULT_QUIESCE_TIMEOUT = 30000; // 30 seconds
 
-export interface ActivityMonitorOptions  { 
-        idleTimeoutMillis : number;
-        quiesceTimeoutMillis : number;
-}
 
+export enum ActivityMonitorStatus {
+    Starting = "starting",
+    Running = "running",
+    Idle = "idle",
+    Stopping = 'stopping'
```

Comment:
Nice idea to use an enum, but by convention, I tried to use UPPER_CASE for enums in typescript